Assignment6.java

```diff
@@ -90,16 +88,17 @@ private static int nextCarToMove(int[] given, int[] desired, Map<Integer, Intege
         
         //car belonging to the current empty slot
         if (desired[indexMap.get(0)] != 0) {
-            return desired[indexMap.get(0)];
+            // remove car from set of misplaced cars
+            int nextCarToMove = desired[indexMap.get(0)];
+            misplacedCars.remove(nextCarToMove);
+            return nextCarToMove;
 
         }
 
         //empty lot in its desired place => deblock the algorithm
         // pick a random car from set of misplaced cars
         if (misplacedCars.size() > 0) {
-            Integer nextCarToMove = misplacedCars.iterator().next();
-            misplacedCars.remove(nextCarToMove);

```

Comment:
If you remove the lines 69-70, then we need to keep here the lines that delete the car from the set of misplaced cars, because this is the case where we unblock the alg (0 on its spot). right now, we only delete the car in the greedy case (line 93). So in brief, we should keep lines 100-102. Am I missing smth?